### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by tsalo**
Closes none. One slightly, but not meaningfully, breaking change is that the `CuBIDS` class is no longer accessible through `cubids` (i.e., you can't do `from cubids import CuBIDS`). Instead, you need the submodule now (`from cubids.cubids import CuBIDS`).

Changes proposed:

- Add a glossary that defines some of the core terms.
- Add API page.
- Split out CLI documentation from general usage walkthrough page.
- Remove duplicate README.rst in docs folder.
- Go through documentation and try to split sentences across lines on punctuation.
- Add a section to README on what CuBIDS is not meant to do.
- Add links from API documentation to GitHub code.

### CODE REVIEW

File: docs/about.rst [hunk 1]
```diff
@@ -1,83 +1,65 @@
...
-        * Key Group: ``acquisition-singleband_datatype-func_suffix-bold_task-rest``
-        * Param Group: ``2`` (Variant Group)
-        * Rename Key Group: ``acquisition-singlebandVARIANTNoFmap_datatype-func_suffix-bold_task-rest``
+
+    *   Example Filename: ``sub-02_ses-A_task-rest_acq-singleband_bold.nii.gz``
+    *   Key Group: ``acquisition-singleband_datatype-func_suffix-bold_task-rest``
+    *   Param Group: ``2`` (Variant Group)
+    *   Rename Key Group: ``acquisition-singlebandVARIANTNoFmap_datatype-func_suffix-bold_task-rest``
 
 In the next section, we'll discuss these definitions in more detail and demonstrate ``CuBIDS`` usage.
```
Comment (mattcieslak) [inline docs/about.rst:64]: Should we link to glossary here?

File: docs/api.rst
Comment (mattcieslak) [inline docs/api.rst:1]: I'd never though of someone using cubids outside of the CLI. Do you think there's a use case for someone using the python functions directly?

File: docs/links.rst
Comment (mattcieslak) [inline docs/links.rst:1]: Great idea to include the link here. The "key group" was something we debated a lot about using and still probably doesn't make sense to all users straight away

File: docs/usage.rst [hunk 2]
```diff
@@ -100,80 +113,96 @@ Modifying Key and Parameter Group Assignments
...
+(i.e., assign them a different BIDS name).
+This can be accomplished by editing the empty columns in the `_summary.csv` file produced by
+``cubids group``.
 
 Once the columns have been edited you can apply the changes to BIDS data using
 
 .. code-block:: console
 
-    $ cubids-apply /bids/dir keyparam_edited new_keyparam_prefix
+    $ cubids apply /bids/dir keyparam_edited new_keyparam_prefix
```
Comment (mattcieslak) [inline docs/usage.rst:126]: this looks way cooler

File: docs/example.rst [hunk 3]
```diff
@@ -1,72 +1,80 @@
...
 you should have successfully installed ``CuBIDS``, ``DataLad``, and the ``bids-validator`` inside a
-conda environment titled "cubids". In this example, we use validator version ``1.7.2``.
-Using a different version of the validator may result in slightly different validation
-tsv outputs, but the example should still be useful. 
-
-Throughout this example, we use ``DataLad`` for version control. Although ``DataLad``
-is an optional dependency of ``CuBIDS``, we use it here to demonstrate its
-powerful integration with ``CuBIDS`` and the benefits it can provide its users. 
+conda environment titled "cubids".
+In this example, we use validator version ``1.7.2``.
```
Comment (mattcieslak) [inline docs/example.rst:18]: how behind is this?

File: docs/example.rst [hunk 4]
```diff
@@ -1,72 +1,80 @@
...
-mkdir $PWD/CuBIDS_Test 
-cd CuBIDS_Test
+Now that we have installed CuBIDS and all necessary dependencies,
+we are ready to begin the curation process on our example dataset.
+We create a ``CuBIDS_Test`` directory to function as our working directory and navigate to it as follows.
 
 .. code-block:: console
 
+    $ mkdir $PWD/CuBIDS_Test
+    $ cd CuBIDS_Test
```
Comment (mattcieslak) [inline docs/example.rst:34]: does this make CuBIDS_test/CuBIDS_Test?

File: docs/example.rst [hunk 5]
```diff
@@ -76,166 +84,179 @@ To do this, we run the following command:
...
-to detect potential curation errors using ``cubids-validate``.
+to detect potential curation errors using ``cubids validate``.
 
 .. code-block:: console
 
-    $ cubids-validate BIDS_Dataset_DataLad v0 --sequential
+    $ cubids validate BIDS_Dataset_DataLad v0 --sequential
 
-.. note::  The use of the ``--sequential`` flag forces the validator to treat each participant as its own BIDS dataset. This can be helpful for identifying heterogeneous elements, but can be slowed down by extremely large datasets.
+.. note::
```
Comment (mattcieslak) [inline docs/example.rst:188]: does this still render correctly?

**Review by mattcieslak — APPROVED** [review]
Excellent!

File: docs/about.rst [hunk 6]
```diff
@@ -1,83 +1,65 @@
...
 
+What CuBIDS Is Not
+------------------
 
+``CuBIDS`` is not designed to convert raw data into BIDS format.
+For that, we recommend using `conversion tools <https://bids.neuroimaging.io/benefits.html#converters>`_.
+``CuBIDS`` then takes over once you have a valid BIDS dataset,
+prior to running any preprocessing or analysis pipelines, or to sharing the dataset.
 
-.. topic:: Key Group
```
[the remark this answers is not in the document]
Comment (tsalo) [reply docs/about.rst:39]: Yeah I was thinking a glossary would be useful. Did you browse the rendered docs? Hoverxref is really cool.

File: docs/api.rst
Comment (tsalo) [reply docs/api.rst:1]: I think there will always be some people who'd rather script their actions with Python, even if it's a small minority.

File: docs/example.rst (hunk 5 above)
Comment (tsalo) [reply docs/example.rst:188]: Yeah everything still shows up in the admonition box

File: docs/usage.rst (hunk 2 above)
Comment (tsalo) [reply docs/usage.rst:126]: I just wish I could figure out a way to link from the subcommand code back to the sphinx-argparse-generated documentation, the way we can do `:func:'cubids.func'`

File: docs/about.rst [hunk 7]
```diff
@@ -1,83 +1,65 @@
...
 
-Definitions
-------------
 
+What CuBIDS Is Not
+------------------
 
+``CuBIDS`` is not designed to convert raw data into BIDS format.
+For that, we recommend using `conversion tools <https://bids.neuroimaging.io/benefits.html#converters>`_.
+``CuBIDS`` then takes over once you have a valid BIDS dataset,
```
[the remark this answers is not in the document]
Comment (tsalo) [reply docs/about.rst:44]: I'll add a note that CuBIDS _should_ work for BIDS-y data, but it's not guaranteed.

File: docs/about.rst (hunk 1 above)
Comment (tsalo) [reply docs/about.rst:64]: Absolutely, will do.

File: docs/example.rst (hunk 3 above)
Comment (tsalo) [reply docs/example.rst:18]: Roughly 3 years old. Probably a good idea to update the example and notebooks at some point.

File: docs/example.rst (hunk 4 above)
Comment (tsalo) [reply docs/example.rst:34]: Ah, I must have pasted it twice. Good catch!

Comment (tsalo) [reply docs/example.rst]: Fixed now.

**Pull request merged**